models/tokenlon/ethereum/tokenlon_v5_ethereum_amm_v1_trades.sql

```diff
@@ -0,0 +1,116 @@
+{{ config(
+    schema = 'tokenlon_v5_ethereum',
+    tags=['dunesql'],
+    alias = alias('amm_v1_trades'),
+    partition_by = ['block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash'],
+    post_hook = '{{ expose_spells(\'["ethereum"]\',
+                                    "project",
+                                    "tokenlon",
+                                    \'["izayl"]\') }}'
+    )
+}} 
+{% set blockchain = 'ethereum' %}
+{% set project = 'tokenlon' %}
+{% set project_version = '5' %}
+-- block 11463016
+{% set project_start_date = "timestamp '2020-12-16'" %}
+
+WITH dexs AS (
+    SELECT 
+        evt_block_time        AS block_time,
+        "userAddr"            AS taker,
+        "makerAddr"           AS maker,
+        "takerAssetAmount"    AS token_sold_amount_raw,
+        "makerAssetAmount"    AS token_bought_amount_raw,
+        CAST(NULL as double)  AS amount_usd,
+        CASE
+            WHEN "takerAssetAddr" IN (0x0000000000000000000000000000000000000000)
+                THEN 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2
+            ELSE "takerAssetAddr" 
+        END                   AS token_sold_address,
+        CASE
+            WHEN "makerAssetAddr" IN (0x0000000000000000000000000000000000000000)
+                THEN 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2
+            ELSE "makerAssetAddr"
+        END                   AS token_bought_address,
+        contract_address      AS project_contract_address,
+        evt_tx_hash           AS tx_hash,
+        ''                    AS trace_address,
```

Comment:
similar question as above, if this feeds `dex.trades`, then you can remove `trace_address` altogether.

if `dex_aggregator`, then we should change this to be an explicit cast to array datatype, for example, on spark:
https://github.com/duneanalytics/spellbook/blob/main/models/oneinch/ethereum/oneinch_v2_ethereum_trades_legacy.sql#L46
we'd do something similar that works on dunesql syntax